### PR TITLE
Add `schema_options` and `field_options` on RowTypeConstraint

### LINKED ISSUE #22083: [Feature Request]: Python native representation for Beam schema options
### What would you like to happen?

Beam schemas have support for options, we should add a native representation for these options in Python. My suggestion would be to normalize on RowTypeConstraint as the Python native representation for a Beam schema and store the options there.

### Issue Priority

Priority: 2

### Issue Component

Component: sdk-py-core

### PR DESCRIPTION
**Pull request by TheNeuralBit**
Fixes #22083. This PR adds a Python-native representation for [Beam schema options](https://cwiki.apache.org/confluence/display/BEAM/%5BBIP-1%5D+Beam+Schema+Options).

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by TheNeuralBit** [comment]
Run Python PreCommit

**Comment by TheNeuralBit** [comment]
@y1chi do you have time to review this? I know the diff is large, but most of it is rote. A lot of it is `switch` statements to handle the supported option types (in implementation and in test), and refactoring tests to be parameterized.

**Review by y1chi — APPROVED** [review]
LGTM, just a few nits.

File: sdks/python/apache_beam/typehints/schemas.py [hunk 1]
```diff
@@ -260,6 +270,86 @@ def typing_to_runner_api(self, type_: type) -> schema_pb2.FieldType:
               representation=self.typing_to_runner_api(
                   logical_type.representation_type())))
 
+  def option_from_runner_api(
+      self, option_proto: schema_pb2.Option) -> Tuple[str, Any]:
+    if not option_proto.HasField('type'):
+      return option_proto.name, None
+
+    fieldtype_proto = option_proto.type
+    if not fieldtype_proto.WhichOneof("type_info") == "atomic_type":
```
[the remark this answers is not in the document]
Comment (TheNeuralBit) [reply sdks/python/apache_beam/typehints/schemas.py]: done

**Pull request merged**